### PR TITLE
Update GRMHD analytic data for use in `WrappedGr`

### PR DESCRIPTION
**Pull request by moxcodes**
## Proposed changes

Adds the necessary type aliases and instantiations so that any GRMHD analytic data or Relativistic Euler analytic solution can be used in `WrappedGr` to provide GH+MHD data.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `major new feature` if appropriate.

### Dependencies

- [x] #3177 (just because of merge conflicts in `WrappedGr`)

### CODE REVIEW

**Comment by moxcodes** [comment]
Updated for the changes in #3177

File: src/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/WrappedGr.hpp [hunk 1]
```diff
@@ -94,6 +94,28 @@ class WrappedGr : public SolutionType {
...
 
+  // overloads for wrapping analytic data
+
+  template <typename... Tags>
+  tuples::TaggedTuple<Tags...> variables(
+      const tnsr::I<DataVector, volume_dim>& x,
+      tmpl::list<Tags...> /*meta*/) const noexcept {
+    // Get the underlying solution's variables using the solution's tags list,
+    // store in IntermediateVariables
+    const IntermediateVars& intermediate_vars = SolutionType::variables(
```
Comment (wthrowe) [inline src/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/WrappedGr.hpp]: I assume `SolutionType::variables` will never return a reference, so remove `&` from the variable declaration.  (Again in the next function.)

**Review by wthrowe — CHANGES_REQUESTED** [review]
The release builds are failing for some reason.

**Comment by moxcodes** [comment]
rebased on develop and posted fixups for the requested change and several updates needed to handle recent changes to the analytic solutions and fix mistakes that I found.
Cheers!

File: src/PointwiseFunctions/AnalyticSolutions/RelativisticEuler/TovStar.cpp [hunk 2]
```diff
@@ -404,15 +412,39 @@ bool operator!=(const TovStar<RadialSolution>& lhs,
...
+      const RadialVariables<DTYPE(data)>& /*radial_vars*/) const noexcept;
+
 #define INSTANTIATE(_, data)                                          \
   template class TovStar<STYPE(data)>;                                \
   template bool operator==(const TovStar<STYPE(data)>& lhs,           \
                            const TovStar<STYPE(data)>& rhs) noexcept; \
   template bool operator!=(const TovStar<STYPE(data)>& lhs,           \
                            const TovStar<STYPE(data)>& rhs) noexcept;
 
+
```
Comment (nilsdeppe) [inline src/PointwiseFunctions/AnalyticSolutions/RelativisticEuler/TovStar.cpp]: drop extra line?

File: src/PointwiseFunctions/AnalyticSolutions/RelativisticEuler/TovStar.cpp [hunk 3]
```diff
@@ -284,6 +284,14 @@ TovStar<RadialSolution>::variables(
   return make_with_value<tnsr::ii<DataType, 3>>(x, 0.0);
 }
 
+template <typename RadialSolution>
+template <typename DataType, typename Tag>
+tuples::TaggedTuple<::Tags::dt<Tag>> TovStar<RadialSolution>::variables(
```
Comment (nilsdeppe) [inline src/PointwiseFunctions/AnalyticSolutions/RelativisticEuler/TovStar.cpp:288]: for my understanding, the `dt` is needed for constructing `Pi`, right?

**Review by nilsdeppe — COMMENTED** [review]
I'm happy for you to squash, including any new changes

**Comment by wthrowe** [comment]
I am happy with the fixups so far.

File: src/PointwiseFunctions/AnalyticSolutions/RelativisticEuler/TovStar.cpp (hunk 3 above)
Comment (moxcodes) [reply src/PointwiseFunctions/AnalyticSolutions/RelativisticEuler/TovStar.cpp:288]: Yeah, that's right; internally, the `WrappedGr` grabs `dt_shift`, `dt_lapse`, and `dt_spatial_metric` and assembles `Pi`.

**Comment by moxcodes** [comment]
fixups squashed in. This will still need a rebase once the dependency is merged in.
Cheers!

**Review by nilsdeppe — DISMISSED** [review]
LGTM

**Review by wthrowe — DISMISSED** [review]

**Comment by nilsdeppe** [comment]
@moxcodes this needs a rebase now :)

**Comment by moxcodes** [comment]
Rebased on develop.
Cheers!

**Review by nilsdeppe — DISMISSED** [review]

**Comment by wthrowe** [comment]
clang-tidy appears to think your code fails to compile, although clang itself disagrees.  I don't know why that would be, but can you check if there is an easy workaround since it strikes me as something that might cause trouble down the road?

**Comment by moxcodes** [comment]
Squashed in a fix that at least solved the clang-tidy problem for me locally. There was an inheritance ambiguity that wasn't really ambiguous because one path was `private` and the other `public`, so the actual compilers didn't mind.
I solved it by just not using the base class factoring and added an explanatory comment for the case that clang-tidy didn't like.
Cheers!

**Review by nilsdeppe — APPROVED** [review]

**Review by wthrowe — APPROVED** [review]

**Pull request merged**